cmd/queue/execprobe.go

```diff
@@ -61,31 +63,32 @@ const (
 // until the HTTP endpoint responds with success. This allows us to get an
 // initial readiness result much faster than the effective upstream Kubernetes
 // minimum of 1 second.
-func standaloneProbeMain(timeout int) (exitCode int) {
-	queueServingPort, err := strconv.Atoi(os.Getenv("QUEUE_SERVING_PORT"))
+func standaloneProbeMain(timeoutSeconds int) (exitCode int) {
```

Comment:
Why not use a time.Duration for this?

Reply:
would require cast and multiplication in main. I tried to avoid redundant work (since if it's 0 — we do nothing).

Reply:
In fact, you know what... we can do it!

Reply:
But it is literally used as `time.Duration` line 79:
```
		timeout = time.Duration(timeoutSeconds) * time.Second
```
BTW, isn't it weird to multiply a `time.Duration` with `time.Second`, what does it even mean.

My point, is that using the right types is always safer. It prevent multiplying bananas with potatoes 😆

Reply:
Ok, done :)

Reply:
Ok, so `time.Duration(timeoutSeconds) * time.Second` actually works but for the wrong reasons. `time.Duration(timeoutSeconds)` ends up being timeoutSeconds **nanoseconds** and then it is multiplied by `time.Seconds` it becomes timeoutSeconds **seconds**. But that's sketchy. Why is this even allowed 😢